src/main/scala/nl/knaw/dans/easy/umd/Command.scala

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.umd
+
+import com.yourmediashelf.fedora.client.FedoraClient
+import com.yourmediashelf.fedora.client.request.FedoraRequest
+import nl.knaw.dans.easy.umd.InputRecord.parse
+import nl.knaw.dans.easy.umd.{CommandLineOptions => cmd}
+import org.slf4j.LoggerFactory
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.transform.{RewriteRule, RuleTransformer}
+import scala.xml.{Elem, Node, Text}
+
+object Command {
+  val log = LoggerFactory.getLogger(getClass)
+
+  def main(args: Array[String]): Unit = {
+    log.debug("Starting command line interface")
+    val ps = cmd.parse(args)
+    FedoraRequest.setDefaultClient(new FedoraClient(ps.fedoraCredentials))
+    run(ps) match {
+      case s: Success[List[Any]] if s.get.nonEmpty => () // logged by run
+      case f: Failure[_] => log.error(s"failed", f.exception)
+      case s: Success[_] => log.info("success")
+    }
+  }
+
+  def run(implicit ps: Parameters): Try[List[Any]] = {
+    for {
+      list <- parse(ps.input)
+      _ = log.info(s"parsed ${list.length} records")
+      failures = list.map(record => (record, update(record))).filter(_._2.isFailure).map(reportError)
```

Comment:
Feels a little uneasy. 

If some uncatched exception prevents execution of the filter and subsequent map, errors on previous input records are not logged. Can't quickly figure out something like recoverWith on the update method.